packages/gafl-webapp-service/src/pages/contact/digital-licence/licence-confirmation-method/__tests__/result-function.spec.js

```diff
@@ -0,0 +1,36 @@
+import resultFunction from '../result-function'
+import { CommonResults } from '../../../../../constants.js'
+
+describe('newsletter > result-function', () => {
+  const getMockRequest = (fromContactDetailsSeen, summary) => ({
+    cache: () => ({
+      helpers: {
+        status: {
+          getCurrentPermission: async () => ({
+            fromContactDetailsSeen: fromContactDetailsSeen,
+            fromSummary: summary
+          })
+        }
+      }
+    })
+  })
+
+  describe('result function', () => {
+    beforeEach(jest.clearAllMocks)
+
+    it('should return amend if status.fromContactDetailsSeen equals seen', async () => {
+      const result = await resultFunction(getMockRequest('seen'))
+      expect(result).toBe(CommonResults.AMEND)
+    })
+
+    it('should return summary if status.fromContactDetailsSeen does not equal seen and status.fromSummary equals contact-summary', async () => {
+      const result = await resultFunction(getMockRequest('not-seen', 'contact-summary'))
+      expect(result).toBe(CommonResults.SUMMARY)
+    })
```

Comment:
There are three conditions that would result in a return value of `CommonResults.SUMMARY`, you should cover all of them.